test/JDBC/expected/TestSpatialFun-vu-verify.out

```diff
@@ -0,0 +1,932 @@
+go
+~~ERROR (Code: 33557097)~~
+
+~~ERROR (Message: Remote procedure/function reference with 4-part object name is not currently supported in Babelfish)~~
+
+
+
+DECLARE @pnt geometry;
+SET @pnt = geometry::STGeomFromText('POINT(-122.34900 47.65100)', 4326);
+select geometry::Point(@pnt.STY, @pnt.STX, 4326).STContains(@pnt)
+DECLARE @referencePoint geometry = geometry::Point(0.0, 0.0, 4326);
+DECLARE @contains BIT = 1;
+SELECT ROUND(PointColumn.STContains(@referencePoint) / @contains, 0) * @contains AS Grp,
+COUNT(*) AS PointCount
+FROM YourTableTemp
+GROUP BY ROUND(PointColumn.STContains(@referencePoint) / @contains, 0) * @contains
+ORDER BY Grp;
+GO
+~~START~~
+bit
+0
+~~END~~
+
+~~START~~
+numeric#!#int
+1#!#2
+~~END~~
+
+
+DECLARE @referencePoint geometry = geometry::Point(1.0, 0.0, 4326);
+SELECT ROUND(PointColumn.STContains(@referencePoint) / @referencePoint.STX, 0) * @referencePoint.STX AS GRP,
+COUNT(*) AS PointCount
+FROM YourTableTemp
+GROUP BY ROUND(PointColumn.STContains(@referencePoint) / @referencePoint.STX, 0) * @referencePoint.STX
+ORDER BY Grp;
+GO
+~~START~~
+float#!#int
+0.0#!#2
+~~END~~
+
+
+DECLARE @referencePoint geometry = geometry::Point(0.0, 0.0, 4326);
+SELECT ID, PointColumn1.STContains(PointColumn2) AS contain,
+cast(PointColumn1.STContains(@referencePoint) as float) - LAG(PointColumn1.STContains(PointColumn2)) OVER (ORDER BY ID) AS Difference 
+FROM YourTableTemp2 ORDER BY PointColumn1.STX;
+GO
+~~ERROR (Code: 33557097)~~
+
+~~ERROR (Message: cannot cast type "bit" to double precision)~~
+
+
+DECLARE @referencePoint geometry = geometry::Point(0.0, 0.0, 4326);
+SELECT ID, PointColumn.STContains(@referencePoint) AS contain,
+cast(@referencePoint.STContains(PointColumn) as float) - LAG(@referencePoint.STX) OVER (ORDER BY ID) AS Difference
+FROM YourTableTemp ORDER BY PointColumn.STX;
+GO
+~~ERROR (Code: 33557097)~~
+
+~~ERROR (Message: cannot cast type "bit" to double precision)~~
+
+
+
+--for STArea
+DECLARE @point geometry;
+SET @point = geometry::STPointFromText('POINT(-122.34900 47.65100)', 4326);
+SELECT STArea(@point);
+go
+~~START~~
+float
+0.0
+~~END~~
+
+
+DECLARE @point geometry;
+SET @point = geometry::STPointFromText('POINT(-122.34900 47.65100)', 4326);
+SELECT @point.STArea();
+Go
+~~START~~
+float
+0.0
+~~END~~
+
+
+DECLARE @point geometry;
+SET @point = geometry::POINT(22.34900, -47.65100, 4326);
+SELECT @point . STArea ( );
+Go
+~~START~~
+float
+0.0
+~~END~~
+
+
+SELECT location.STArea() from SPATIALPOINTGEOM_dttemp ORDER BY location.STX;
+GO
+~~START~~
+float
+<NULL>
+0.0
+0.0
+0.0
+0.0
+0.0
+0.0
+~~END~~
+
+
+DECLARE @point geography;
+SET @point = geography::STGeomFromText('POINT(-122.34900 47.65100)', 4326);
+SELECT STArea(@point);
+SELECT @point.STArea();
+Go
+~~START~~
+float
+0.0
+~~END~~
+
+~~START~~
+float
+0.0
+~~END~~
+
+
+DECLARE @point geography;
+SET @point = geography::Point(22.34900, -47.65100, 4326);
+SELECT STArea(@point);
```

Comment:
Currently we support `Point` shape so we can't verify the precision/scale with T-SQL but we should make sure that our solution can incorporate such changes in future if needed and not create a one way door so that we are unable to fix such issues.

Reply:
we have used PostGIS functions only to calculate the area, although it's true that we can't verify the precision from TSQL now, but other geometries can be incorporated using this function in future, and we can document minor precision differences if any, as we did for STDistance